ui/app/src/components/Header.tsx

```diff
@@ -11,15 +11,108 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import { AppBar, Switch, Toolbar, Typography } from '@material-ui/core';
+import {
+  AppBar,
+  Box,
+  Button,
+  CircularProgress,
+  Divider,
+  Menu,
+  MenuItem,
+  Switch,
+  Theme,
+  Toolbar,
+  Typography,
+} from '@mui/material';
+import { ChevronDown } from 'mdi-material-ui';
+import { SxProps } from '@mui/system/styleFunctionSx/styleFunctionSx';
+import { MouseEvent, useState } from 'react';
+import { ProjectModel, useProjectQuery } from '../model/project-client';
+import Toast from './Toast';
+
+function ProjectMenuComponent(props: {
+  projectList: ProjectModel[];
+}): JSX.Element {
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+
+  const handleMenu = (event: MouseEvent<HTMLElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+  const handleCloseMenu = () => {
+    setAnchorEl(null);
+  };
+  return (
+    <>
+      <Button
+        aria-label="List of the available projects"
+        aria-controls="menu-project-list-appbar"
+        aria-haspopup="true"
+        color="inherit"
+        endIcon={<ChevronDown />}
+        onClick={handleMenu}
+      >
+        Projects
+      </Button>
+      <Menu
+        id="menu-project-list-appbar"
+        anchorEl={anchorEl}
+        anchorOrigin={{
+          vertical: 'bottom',
+          horizontal: 'left',
+        }}
+        keepMounted
+        open={anchorEl !== null}
+        onClose={handleCloseMenu}
+      >
+        {props.projectList.map((value, index) => {
+          return (
+            // TODO when routing is in place, use the button to redirect to the project page
+            <MenuItem key={index} onClick={handleCloseMenu}>
+              {value.metadata.name}
+            </MenuItem>
+          );
+        })}
+      </Menu>
+    </>
+  );
+}
+
+function ProjectMenu(): JSX.Element {
```

Comment:
I think you can just inline the `ProjectMenuComponent` directly into this one. No need to have them be separate. You can also just use an early return for the loading indicator (pretty common pattern in React). For example:

```tsx
if (isLoading) {
  return <CircularProgress />;
}

// Regular rendering after loading has finished...
return (
  <>
    <Button />
    { /* ... rest of regular JSX markup ... */
  </>
);
```

Would also be cool to break it out into its own file, but don't feel like you have to if you like it next to `Header`. Get the feeling we might end up doing it eventually if this header keeps growing.

Reply:
ah yeah that's cleaner. will do that. Thanks for the tips !

Reply:
> Would also be cool to break it out into its own file
yeah probably it will make sense once we would add another component in the `Header`